diesel_derives/src/identifiable.rs

```diff
@@ -21,9 +23,19 @@ pub fn derive(item: DeriveInput) -> Result<TokenStream> {
     let mut field_ty = Vec::new();
     let mut field_name = Vec::new();
     for pk in model.primary_key_names.iter() {
-        let f = model.find_column(pk)?;
-        field_ty.push(&f.ty);
-        field_name.push(&f.name);
+        let Field {
+            ty,
+            name,
+            serialize_as,
+            ..
+        } = &model.find_column(pk)?;
+        if let Some(AttributeSpanWrapper { item: ty, .. }) = serialize_as.as_ref() {
+            field_ty.push(quote!(#ty));
+            field_name.push(quote!(::std::convert::Into::<#ty>::into(self.#name)));
```

Comment:
I wonder if we can change this to 
```suggestion
            field_name.push(quote!(::std::convert::Into::<#ty>::into(self.#name.clone())));
```

This would at least relax the `Copy` requirement and `Clone` is automatically implemented for any `Copy` type. The other alternative would be to require a `From<&Key>` impl instead.

Reply:
Shifting the requirement to `Clone` instead will silently clone objects that might have a non-trivial `Clone` implementation (e.g. `Vec`).

I'm not closed to the idea if you think it's best, but I'll wait for other people to review, as per your other comment (I'm not sure it pinged anyone though)

Reply:
Fair enough, and done. Do you want me to write a test-case for that?

Reply:
That would be great to have.